src/GraphQL/Validation/DocumentValidator.cs

```diff
@@ -18,8 +22,12 @@ public interface IDocumentValidator
             Inputs inputs = null);
     }
 
+    /// <inheritdoc/>
     public class DocumentValidator : IDocumentValidator
     {
+        /// <summary>
+        /// Returns the default set of validation rules: all except <see cref="OverlappingFieldsCanBeMerged"/>.
```

Comment:
How hard is this rule? Is everything so bad with it?

Reply:
I don't know.  But it's probably twice as much code as all the rest of the validation rules combined.  Creates a bunch of lists and stuff.  Take a look.  I figured that for 3.3 we probably wouldn't be changing the core set of rules, and in 4.0 we can decide if that behavior should be changed.

Reply:
But at least it is obvious now that there's a validation rule that is not in use.

Reply:
Let's make new issue about this rule and keep it as is for now.